Schedule, Schedule entries, Trigger, Acknowledge, and Log Entries

lszomoru opened the pull request:
I was wondering if you would be willing to accept my pull request which contains wrappers for the Schedule, Schedule entries, Trigger, and Acknowledge REST APIs in order to extend the capabilities of the library. If you are willing to accept the pull request it would be great if you could post an updated package to Nuget. Do let me know if there are any changes to the code that you want me to make in order to accept the pull request. Thanks!

usbsnowcrash commented:
I'm traveling this weekend but should be able to look Sunday or Monday

lszomoru commented:
No worries. Actually I just realized that I would like to add another REST API wrapper for getting the log entries for a particular incident. Let me work on that and I will update the pull request.

lszomoru commented:
I have added the wrappers for the Log Entries. Please review the code and let me know your feedback.  Thanks!

usbsnowcrash commented on Source/PagerDuty.Net.Tests/Models/IncidentsResponseTests.cs:
```diff
@@ -30,12 +30,12 @@ public class IncidentsResponseTests {
 
             Assert.IsNotNull(myAlert);
             Assert.IsNotNull(myAlert.incidents);
+            Assert.IsNotNull(myAlert.incidents);
```
Why do we assert this twice?

lszomoru replied:
Fixed.

usbsnowcrash commented on Source/PagerDuty.Net.Tests/Models/AlertTests.cs:
```diff
@@ -33,7 +33,7 @@ public class AlertTests {
             Assert.AreEqual("Phone", myAlert.type);
             Assert.AreEqual("+15555551234", myAlert.address);
-            Assert.AreEqual(new DateTime(634981805300000000), myAlert.started_at);
+            Assert.AreEqual(new DateTime(634982021300000000), myAlert.started_at);
```
I didn't realize my test was failing in master!  thanks!

lszomoru replied:
No problem!

usbsnowcrash commented on Source/PagerDuty.Net/Models/Channel.cs:
```diff
@@ -0,0 +1,42 @@
+
+        //Channel specific fields
+
+        //Nagios, API, Web trigger channel specific
```
What does Nagios mean?

lszomoru replied:
https://www.pagerduty.com/docs/guides/nagios-integration-guide/

usbsnowcrash commented on Source/PagerDuty.Net/Models/Channel.cs:
```diff
@@ -0,0 +1,42 @@
+        //Nagios 
+        public string host { get; set; }
+        public string service { get; set; }
+        public string state { get; }
```
needs bot a get and a set

lszomoru replied:
Fixed.

usbsnowcrash commented on Source/PagerDuty.Net/PagerDutyAPI.cs:
```diff
@@ -228,6 +346,80 @@ public class PagerDutyAPI {
+        /// <param name="integrationUrl">Integration API Url</param>
+        /// <param name="aRequest">Trigger details</param>
+        /// <returns></returns>
+        public IntegrationResponse Trigger(string integrationUrl, TriggerRequest tRequest) {
```
Why have them pass in the integrationurl?  When you instantiate the PagerDuty api wrapper you pass in the subdomain pointing to your account.  If you look at this https://github.com/usbsnowcrash/PagerDuty.Net/blob/master/Source/PagerDuty.Net/PagerDutyAPI.cs#L34 you can see how I do it for the other endpoints.

usbsnowcrash commented on Source/PagerDuty.Net/PagerDutyAPI.cs:
```diff
@@ -47,6 +47,16 @@ public class PagerDutyAPI {
+        /// <param name="integrationUrl">Integration API Url</param>
+        /// <param name="aRequest">Acknowledge details</param>
+        /// <returns></returns>
+        public IntegrationResponse Acknowledge(string integrationUrl, AcknowledgeRequest aRequest) {
```
Passing in an integration url does not make sense I think.  It looks like (from the pagerduty api docs) that you always post to https://events.pagerduty.com/generic/2010-04-15/create_event.json for the integration apis.

lszomoru replied:
Fixed.

lszomoru commented:
Not really sure about the AppVeyor build failure, as the two tests that fail are passing on the latest code.

usbsnowcrash commented:
I see the same tests failing on my machine as they do in appveyor.

![screenshot 2015-07-20 14 08 23](https://cloud.githubusercontent.com/assets/882850/8783282/d68c1d2c-2ee8-11e5-9e9e-81789796b934.png)
I think we need to change the asserts in these two tests to be something like this
```cs
Assert.AreEqual(DateTime.Parse("2013-03-06T15:28:50-05:00"), myAlert.started_at);
```
Once these are passing I will merge :zap:

lszomoru commented:
Strange. Let me take a look and update the pull request.

lszomoru commented:
Fixed. I think I know what the problem was. Those two tests were using JSON files that had time zone specific timestamps in them. Due to that the tests were failing on my machine. After I have fixed them, they were passing on my machine but were failing on the build and your machine. The fix was to switch to a UTC timestamp.

lszomoru commented:
Thanks for merging in my changes. Are you planning on publishing the new version to Nuget?

usbsnowcrash commented:
@lszomoru already published, just takes a few min for it to show up https://www.nuget.org/packages/PagerDutyNet/

lszomoru commented:
@usbsnowcrash, awesome! Thanks! Please feel free to ping me if anyone reports any issues with the newly added wrappers. I will continue to look at PagerDuty and will post a new pull request in case I am adding new functionality.
